### ISSUE TITLE
Get list of Tabs for a template

### ISSUE THREAD

**Issue by miketonguk** [docusign/docusign-csharp-client#129]
I want to return all the tabs for a template by recipient, which I can achieve with the following REST request:

restapi/v2/accounts/{accountId}/envelopes/{templateId}/recipients?include_tabs=true&include_extended=true

Which API method do I use for this, I've searched through the code but couldn't find a method for this request type.

**Comment by miketonguk** [comment]
I found it, the reason I missed it was that the templateId is called a envelopeId in the API. Im guessing that maybe this accepts either a tempalteId or an envelopeId? If so, the parameter should be renamed accordingly.

However! The method doesn't work. Using postman I get the desired response, however when calling via the API I get a JSON parsing error:

```
Data: {System.Collections.ListDictionaryInternal}
    ErrorCode: 500
    ErrorContent: null
    HResult: -2146233088
    HelpLink: null
    InnerException: null
    Message: "Unexpected character encountered while parsing value: {. Path 'signers[0].tabs.dateTabs', line 56, position 11."
    Source: "DocuSign.eSign"
    StackTrace: "   at DocuSign.eSign.Client.ApiClient.Deserialize(IRestResponse response, Type type)\r\n   at DocuSign.eSign.Api.EnvelopesApi.ListRecipientsWithHttpInfo(String accountId, String envelopeId, ListRecipientsOptions options)\r\n   at DocuSign.eSign.Api.EnvelopesApi.ListRecipients(String accountId, String envelopeId, ListRecipientsOptions options)"
    TargetSite: {System.Object Deserialize(RestSharp.IRestResponse, System.Type)}
```

For reference, here's the JSON for the same call via postman:

```
"dateTabs": [
          {
            "validationPattern": "",
            "validationMessage": "",
            "shared": "false",
            "requireInitialOnSharedChange": "false",
            "requireAll": "false",
            "name": "someDate1ToolTip",
            "value": "",
            "width": 168,
            "required": "true",
            "locked": "false",
            "concealValueOnDocument": "false",
            "disableAutoSize": "false",
            "tabLabel": "someDate1",
            "bold": "true",
            "italic": "false",
            "underline": "false",
            "fontSize": "size24",
            "documentId": "1",
            "recipientId": "1",
            "pageNumber": "1",
            "xPosition": "43",
            "yPosition": "183",
            "tabId": "e68d65d6-e280-4e64-8a30-902c6b1eeb63",
            "templateLocked": "false",
            "templateRequired": "true",
            "tabOrder": "0"
          }
        ]
```

I also ran the entire JSON response through a JSON validator and it's all good. Must be some issue with your custom deserialiser in the c# sdk.

**Comment by brbeaird** [comment]
I'm not quite sure what you're wanting to do. Are you wanting to see the tabs of an actual document that was completed by a recipient? Or are you wanting to see the empty placeholder tabs of a template that would be used to populate instances of that template?

Based on your original equivalent endpoint, you're wanting the 2nd option. However, the SDK method you're referring to actually does the 1st option (that's why it takes EnvelopeID and not TemplateId, because it's actually pulling down tabs from an actual document that was signed by a recipient) - I use this all the time, and it works perfectly for pulling out data the recipient entered.

I honestly don't know if the SDK supports pulling down tabs for a template that aren't actually tied to a real document.

**Comment by miketonguk** [comment]
Yes, you are correct, I want to get a list of all the tabs within a template so I know what values are expected to be sent in the envelope to populate a document.

I found the endpoint I was using from this stack overflow response:

http://stackoverflow.com/questions/20249190/getting-all-tabs-from-a-template-using-rest-docusign-api

And it does appear to do exactly what I need, but I agree it doesn't seem right to be passing in a templateId where the api appears to expect an envelopeId. But it does work (via the rest api) just not when using the sdk.

 Is there's a better way to achieve the same goal?

**Comment by brbeaird** [comment]
That's an interesting method I hadn't thought of. 

We have one table that holds all merge-able template fields across all of our templates (about 50 fields total) along with the query that needs to be run to get the value. Whenever we generate a new doc from a template, we actually send ALL of those fields with their values. It seems kind of overkill, but DocuSign handles it well and only grabs the fields that apply to that current doc and throws away all the others.

It keeps it easy on us by not having to track (or look up every time) which template fields apply to which template.

**Comment by gsnavin** [comment]
@miketonguk This issue with the dateTabs should now be fixed with version 2.1.7. Please try it out.
